### DISCUSSION TITLE
Are there plans on supporting import.meta?

### DISCUSSION THREAD

**Discussion by jvliwanag** [facebook/hermes#1540] [Q&A]
ESM builds of a number of libraries depend on `import.meta`. As `metro` also has support for reading exports (albeit unstable), and if enabled, the default config from expo picks up ESM, this becomes an issue. Support for `import.meta` cannot be checked at runtime since hermes blocks its use completely.

**Comment by tmikov** [comment]
Hi, I am not sure I fully understand the question. The module system is provided by Metro, not by Hermes, so Metro needs to support it.

For some context why the module system cannot be provided by Hermes, please see the discussion here: https://github.com/facebook/hermes/issues/1382.

So, even if Hermes did support `import.meta`, there would be no way to use it in RN, at least in the short term.

**Comment by jvliwanag** [comment]
Ah, so stuff like module resolution should be provided outside of Hermes, so that's out.

As a start however, would it be possible to just allow references to `import.meta` even as an empty null-prototype object work? That way code such as:

```
return (import.meta.env ? import.meta.env.MODE : void 0) !== "production"
```

will continue to work, instead of the current hermes behavior of erroring out [here](https://github.com/facebook/hermes/blob/f0bda9338da8228a2aaf4f5b51bb49a824c0bfff/lib/AST/SemanticValidator.cpp#L139).

**Reply by ljharb** [reply]
The syntax is only legal in Module code, and Hermes only supports Script.

**Reply by tmikov** [reply]
@jvliwanag What is the desired behavior for Hermes here?
